s3tests_boto3/functional/test_iam.py

```diff
@@ -413,3 +419,300 @@ def test_delete_user_policy_from_multiple_policies():
     client.delete_user_policy(PolicyName='AllowAccessPolicy2', UserName=get_tenant_user_id())
     client.get_user_policy(PolicyName='AllowAccessPolicy3', UserName=get_tenant_user_id())
     client.delete_user_policy(PolicyName='AllowAccessPolicy3', UserName=get_tenant_user_id())
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow Bucket Actions in user Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_allow_bucket_actions_in_user_policy():
+    client = get_tenant_iam_client()
+    bucket = get_new_bucket()
+    bucket_listed = False
+
+    policy_document_allow = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": ["s3:ListAllMyBuckets", "s3:DeleteBucket"],
+             "Resource": "arn:aws:s3:::*"}}
+    )
+    client.put_user_policy(PolicyDocument=policy_document_allow, PolicyName='AllowAccessPolicy',
+                           UserName=get_tenant_user_id())
+    s3_client = get_client()
```

Comment:
1. Please attach user policy to a different user using the iam_client.
2. s3_client is not created using the same user_id to which the user policy is attached (correct me if I am wrong), so is it being really tested if the user policy is exercised?
3. In order to create s3_client using creds of the user to which the user policy is attached, please see: https://github.com/ceph/s3-tests/blob/master/s3tests_boto3/functional/test_sts.py#L90

Reply:
yes, we will use the same user(on which user policy is applied) to create s3_client.